htmresearch/algorithms/column_pooler.py

```diff
@@ -273,91 +281,68 @@ def _computeInferenceMode(self, feedforwardInput, lateralInputs):
         lateralInput, self.connectedPermanenceDistal)
       numActiveSegmentsByCell[overlaps >= self.activationThresholdDistal] += 1
 
-    # Activate some of the feedforward supported cells
-    minNumActiveCells = self.sdrSize / 2
-    chosenCells = self._chooseCells(feedforwardSupportedCells,
-                                    minNumActiveCells, numActiveSegmentsByCell)
+    chosenCells = []
+    minNumActiveCells = int(self.sdrSize * .75)
 
-    # If necessary, activate some of the previously active cells
-    if len(chosenCells) < minNumActiveCells:
-      remaining = numpy.setdiff1d(prevActiveCells, feedforwardSupportedCells)
-      remaining = remaining[numActiveSegmentsByCell[remaining] > 0]
+    numActiveSegsForFFSuppCells = numActiveSegmentsByCell[
+        feedforwardSupportedCells]
 
-      chosenCells = numpy.append(
-        chosenCells, self._chooseCells(remaining,
-                                       minNumActiveCells - len(chosenCells),
-                                       numActiveSegmentsByCell))
+    # First, activate the FF-supported cells that have the highest number of
+    # lateral active segments (as long as it's not 0)
+    if len(feedforwardSupportedCells) == 0:
+      pass
+    else:
+      # This loop will select the FF-supported AND laterally-active cells, in
+      # order of descending lateral activation, until we exceed the
+      # minNumActiveCells quorum - but will exclude cells with 0 lateral
+      # active segments.
+      ttop = numpy.max(numActiveSegsForFFSuppCells)
+      while ttop > 0 and len(chosenCells) <= minNumActiveCells:
+        chosenCells = numpy.union1d(chosenCells,
+                    feedforwardSupportedCells[numActiveSegsForFFSuppCells >
+                    self.distalSegmentInhibitionFactor * ttop])
+        ttop -= 1
+
+    # If we still haven't filled the minNumActiveCells quorum, add in the
+    # FF-supported cells with 0 lateral support AND the inertia cells.
+    if len(chosenCells) < minNumActiveCells:
+      if self.useInertia:
+        prevCells = numpy.setdiff1d(prevActiveCells, chosenCells)
+        inertial_cap = int(len(prevCells) * self.inertiaFactor)
+        if inertial_cap > 0:
```

Comment:
I like the `snake_case`, but we should stay consistent with the surrounding code. `inertialCap`, `sortIndices`.